backend/avataxApi.ts

```diff
@@ -0,0 +1,73 @@
+import Avatax from "ava-typescript";
+import { AvataxConfig, FetchTaxesPayload } from "./types";
+import { DocumentType } from "ava-typescript/lib/enums/DocumentType";
+
+const getAvataxClient = (avataxConfig: AvataxConfig) =>
+  new Avatax({
+    appName: "Saleor",
+    appVersion: "1.0",
+    environment: avataxConfig.sandbox ? "sandbox" : "production",
+    machineName: "Saleor-Avatax",
+    timeout: 4000,
+  }).withSecurity({
+    username: avataxConfig.account,
+    password: avataxConfig.password,
+  });
+
+export const fetchTaxes = async (
+  taxPayload: FetchTaxesPayload,
+  avataxConfig: AvataxConfig,
+  transactionType: string
+) => {
+  const type =
+    transactionType === "SalesInvoice"
+      ? DocumentType.SalesInvoice
+      : DocumentType.SalesOrder;
+  const client = getAvataxClient(avataxConfig);
+  const address = taxPayload.address!;
+  const response = await client.createTransaction({
+    model: {
+      type: type,
+      companyCode: avataxConfig.companyCode,
+      code: taxPayload.code || "",
+      date: new Date(taxPayload.date),
+      customerCode: "0",
+      discount: taxPayload.discount,
+      commit: false,
+      currencyCode: taxPayload.currency,
+      // @ts-ignore Avatax typing requires unneeded fields as mandatory.
```

Comment:
Yikes. Using `ts-ignore` is nuclear option - could we instead re-declare Avatax types so that they use optional fields instead?